### PR TITLE
update to v1.0.4 doc

### PR DESCRIPTION
**Pull request by maxbriel**
1. Includes updated installation instructions for v1.0.4 from [PR#450](https://github.com/POSYDON-code/POSYDON/pull/450) in the POSYDON-code repo.
2. Changes the path to v1.0.4 for the documentation
3. Change all the page titles to v1.0.4 instead of 1.0.0-dirty-hash.
4. Although it looks like some pages are deleted in the changes, the files are there! Github's diff is not working properly because the files are all very similar.
5. Changes the copyright to 2024.
6. Some additional API pages were added. These are automatically generated and missing beforehand.

### CODE REVIEW

File: docs/v1.0.4/api/posydon.active_learning.html [hunk 1]
```diff
@@ -1,21 +1,20 @@
 <!DOCTYPE html>
 <html class="writer-html5" lang="en" >
 <head>
-  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />
+  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
```
Comment (mkruckow) [inline docs/v1.0.4/api/posydon.active_learning.html:4]: Just to double check, is it intended to go to an older version here?

File: docs/v1.0.4/visualization/plot2D/plot2D.html [hunk 2]
```diff
@@ -129,7 +128,7 @@
            <div itemprop="articleBody">
              
   <section id="d-plotting-functionalities-for-posyon-grids">
-<span id="plot2d"></span><h1>2D plotting functionalities for POSYON GRIDS<a class="headerlink" href="#d-plotting-functionalities-for-posyon-grids" title="Link to this heading"></a></h1>
+<span id="plot2d"></span><h1>2D plotting functionalities for POSYON GRIDS<a class="headerlink" href="#d-plotting-functionalities-for-posyon-grids" title="Permalink to this headline"></a></h1>
```
Comment (elizabethteng) [inline docs/v1.0.4/visualization/plot2D/plot2D.html:131]: typo in POSYDON

File: docs/v1.0.4/_modules/posydon/binary_evol/step_end.html [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+<li class="toctree-l1"><a class="reference internal" href="../../../pop_synth/custom_flow.html">Custom population synthesis</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../SingleStar/SingleStar.html">The SingleStar object</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../BinaryStar/BinaryStar.html">The BinaryStar object</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../advanced_pop_options/custom_hooks.html">Evolutionary Hooks</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../advanced_pop_options/debugging_binaries.html">Debugging Failed Binaries</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../interpolation/IF-Interpolator/IFInterpolator.html">Initial-Final Interpolation</a></li>
+</ul>
+<p class="caption" role="heading"><span class="caption-text">Plotting</span></p>
+<ul>
+<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot1D/plot1D.html">1D plotting functionalities for POSYON GRIDS</a></li>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/binary_evol/step_end.html:69]:
```suggestion
<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot1D/plot1D.html">1D plotting functionalities for POSYDON GRIDS</a></li>
```

File: docs/v1.0.4/_modules/posydon/binary_evol/step_end.html [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot2D/plot2D.html">2D plotting functionalities for POSYON GRIDS</a></li>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/binary_evol/step_end.html:70]:
```suggestion
<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot2D/plot2D.html">2D plotting functionalities for POSYDON GRIDS</a></li>
```

File: docs/v1.0.4/_modules/posydon/grids/downsampling_report.html [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+<li class="toctree-l1"><a class="reference internal" href="../../../pop_synth/custom_flow.html">Custom population synthesis</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../SingleStar/SingleStar.html">The SingleStar object</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../BinaryStar/BinaryStar.html">The BinaryStar object</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../advanced_pop_options/custom_hooks.html">Evolutionary Hooks</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../advanced_pop_options/debugging_binaries.html">Debugging Failed Binaries</a></li>
+<li class="toctree-l1"><a class="reference internal" href="../../../interpolation/IF-Interpolator/IFInterpolator.html">Initial-Final Interpolation</a></li>
+</ul>
+<p class="caption" role="heading"><span class="caption-text">Plotting</span></p>
+<ul>
+<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot1D/plot1D.html">1D plotting functionalities for POSYON GRIDS</a></li>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/downsampling_report.html:69]:
```suggestion
<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot1D/plot1D.html">1D plotting functionalities for POSYDON GRIDS</a></li>
```

File: docs/v1.0.4/_modules/posydon/grids/downsampling_report.html [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot2D/plot2D.html">2D plotting functionalities for POSYON GRIDS</a></li>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/downsampling_report.html:70]:
```suggestion
<li class="toctree-l1"><a class="reference internal" href="../../../visualization/plot2D/plot2D.html">2D plotting functionalities for POSYDON GRIDS</a></li>
```

File: docs/v1.0.4/_modules/posydon/binary_evol/step_end.html (hunk 3 above)
Comment (sgossage) [reply docs/v1.0.4/_modules/posydon/binary_evol/step_end.html:69]: this seems like a recurrent typo throughout various `.html` files that we may consider correcting

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 7]
```diff
@@ -0,0 +1,546 @@
...
+<span class="p">}</span>
+
+
+<div class="viewcode-block" id="post_process_grid"><a class="viewcode-back" href="../../../api/posydon.grids.post_processing.html#posydon.grids.post_processing.post_process_grid">[docs]</a><span class="k">def</span> <span class="nf">post_process_grid</span><span class="p">(</span><span class="n">grid</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">star_2_CO</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">MODEL</span><span class="o">=</span><span class="n">MODEL</span><span class="p">,</span>
+                      <span class="n">single_star</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
+<span class="w">    </span><span class="sd">&quot;&quot;&quot;Compute post processed quantitiy of any grid.</span>
+
+<span class="sd">    This function post process any supported grid and computes:</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:154]:
```suggestion
<span class="sd">    This function post processes any supported grid and computes:</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 8]
```diff
@@ -0,0 +1,546 @@
...
+<span class="sd">      A: direct collapse</span>
+<span class="sd">      B: Fryer+12-rapid</span>
+<span class="sd">      C: Fryer+12-delayed</span>
+<span class="sd">      D: Sukhbold+16-engine, N20</span>
+<span class="sd">      E: Patton&amp;Sukhbold20-engine, N20</span>
+<span class="sd">      for each prescrition we store the compact object state (WD/NS/BH),</span>
+<span class="sd">      SN type (WD, ECSN, CCSN, PISN, PPISN), fallback mass fraction f_gb,</span>
+<span class="sd">      compact object mass and spin.</span>
+<span class="sd">    - Core masses at he-depletion (used in Patton core collpase)</span>
+<span class="sd">    - Mass envelopes for common ennvelope step.</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:166]:
```suggestion
<span class="sd">    - Mass envelopes for common envelope step.</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 9]
```diff
@@ -0,0 +1,546 @@
...
+<span class="sd">    assumption given in MODEL plus:</span>
+<span class="sd">      A: direct collapse</span>
+<span class="sd">      B: Fryer+12-rapid</span>
+<span class="sd">      C: Fryer+12-delayed</span>
+<span class="sd">      D: Sukhbold+16-engine, N20</span>
+<span class="sd">      E: Patton&amp;Sukhbold20-engine, N20</span>
+<span class="sd">      for each prescrition we store the compact object state (WD/NS/BH),</span>
+<span class="sd">      SN type (WD, ECSN, CCSN, PISN, PPISN), fallback mass fraction f_gb,</span>
+<span class="sd">      compact object mass and spin.</span>
+<span class="sd">    - Core masses at he-depletion (used in Patton core collpase)</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:165]:
```suggestion
<span class="sd">    - Core masses at he-depletion (used in Patton core collapse)</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 10]
```diff
@@ -0,0 +1,546 @@
...
+                      <span class="n">single_star</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
+<span class="w">    </span><span class="sd">&quot;&quot;&quot;Compute post processed quantitiy of any grid.</span>
+
+<span class="sd">    This function post process any supported grid and computes:</span>
+<span class="sd">    - Core collpase quantities for 5 prescritions given the fiducial POSYDON</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:155]:
```suggestion
<span class="sd">    - Core collapse quantities for 5 prescriptions given the fiducial POSYDON</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 11]
```diff
@@ -0,0 +1,546 @@
...
+<span class="sd">      for each prescrition we store the compact object state (WD/NS/BH),</span>
+<span class="sd">      SN type (WD, ECSN, CCSN, PISN, PPISN), fallback mass fraction f_gb,</span>
+<span class="sd">      compact object mass and spin.</span>
+<span class="sd">    - Core masses at he-depletion (used in Patton core collpase)</span>
+<span class="sd">    - Mass envelopes for common ennvelope step.</span>
+
+<span class="sd">    Parameters</span>
+<span class="sd">    ----------</span>
+<span class="sd">    grid : PSyGrid</span>
+<span class="sd">        MESA gri in PSyGrid format.</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:171]:
```suggestion
<span class="sd">        MESA grid in PSyGrid format.</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 12]
```diff
@@ -0,0 +1,546 @@
...
+<span class="sd">        If `True` print on screen the results of each core collpase.</span>
+
+<span class="sd">    Returns</span>
+<span class="sd">    -------</span>
+<span class="sd">    MESA_dirs: list</span>
+<span class="sd">        List containing the path to each run corresponding to the post</span>
+<span class="sd">        processed values. This is used to ensure one to one mapping when</span>
+<span class="sd">        appending the extra columns back to a grid.</span>
+<span class="sd">    EXTRA_COLUMNS: dict</span>
+<span class="sd">        Dictionary containing all post processe quantities.</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:191]:
```suggestion
<span class="sd">        Dictionary containing all post processed quantities.</span>
```

File: docs/v1.0.4/_modules/posydon/grids/post_processing.html [hunk 13]
```diff
@@ -0,0 +1,546 @@
...
+<span class="sd">    grid : PSyGrid</span>
+<span class="sd">        MESA grid in PSyGrid format.</span>
+<span class="sd">    MESA_dirs: list</span>
+<span class="sd">        List containing the path to each run corresponding to the post</span>
+<span class="sd">        processed values. This is used to ensure one to one mapping when</span>
+<span class="sd">        appending the extra columns back to a grid.</span>
+<span class="sd">    EXTRA_COLUMNS: dict</span>
+<span class="sd">        Dictionary containing all post processe quantities.</span>
+<span class="sd">    verbose : bool</span>
+<span class="sd">        If `True` print on screen the results of each core collpase.</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/post_processing.html:485]:
```suggestion
<span class="sd">        If `True` print the results of each core collapse on screen.</span>
```

File: docs/v1.0.4/_modules/posydon/grids/psygrid.html [hunk 14]
```diff
@@ -0,0 +1,2175 @@
...
+
+<span class="sd">    mygrid = PSyGrid().create(in_dir_path, out_file_path,</span>
+<span class="sd">                              overwrite=True,</span>
+<span class="sd">                              warn=&quot;suppress&quot;,</span>
+<span class="sd">                              max_number_of_runs=1000,</span>
+<span class="sd">                              description=&quot;My 1000-run grid.&quot;)</span>
+
+<span class="sd">    NOTE: you can use a family of HDF5 files in case of file size restrictions.</span>
+<span class="sd">    For example, using `mygrid.h5.%d` for the path when creating/loading grids,</span>
+<span class="sd">    implied that split files will be `mygrid.h5.0`, `mygrid.h5.1`, and so on.</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/psygrid.html:133]:
```suggestion
<span class="sd">    implies that split files will be `mygrid.h5.0`, `mygrid.h5.1`, and so on.</span>
```

File: docs/v1.0.4/_modules/posydon/grids/psygrid.html [hunk 15]
```diff
@@ -0,0 +1,2175 @@
...
+
+<span class="sd">    for run in mygrid:</span>
+<span class="sd">        do_something_with(run.history1)</span>
+
+<span class="sd">the table is loaded from the HDF5 file temporarily. This means that:</span>
+
+<span class="sd">    myrun.history1.star_mass + myrun.history1.star_mass</span>
+
+<span class="sd">loads the table two times! To reduce reading times, store in local variables</span>
+<span class="sd">all the tables that are going to be needed more than once:</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/psygrid.html:267]:
```suggestion
<span class="sd">loads the table two times! To reduce reading times, store all the tables</span>
<span class="sd">that are going to be needed more than once in local variables:</span>
```

File: docs/v1.0.4/_modules/posydon/grids/psygrid.html [hunk 16]
```diff
@@ -0,0 +1,2175 @@
...
+
+<span class="sd">    with open(&quot;grid_summary.txt&quot;, &quot;w&quot;) as f:</span>
+<span class="sd">        f.write(str(mygrid))</span>
+
+<span class="sd">b) In `for` loop (`iter` and `next` are supported too):</span>
+
+<span class="sd">    for run in mygrid:</span>
+<span class="sd">        print(run)</span>
+
+<span class="sd">c) Getting number of runs:</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/psygrid.html:232]:
```suggestion
<span class="sd">c) Getting the number of runs:</span>
```

File: docs/v1.0.4/_modules/posydon/grids/psygrid.html [hunk 17]
```diff
@@ -0,0 +1,2175 @@
...
+<span class="sd">   the form `*_saved_columns`. For example,</span>
+
+<span class="sd">       grid.create(&quot;mygrid/&quot;, &quot;mygrid.h5&quot;,</span>
+<span class="sd">                   star1_history_saved_columns=[&quot;star_mass&quot;])</span>
+
+<span class="sd">   By default, the provided columns will be added to the default list. In order</span>
+<span class="sd">   to select the exact set of columns, provide them in a tuple instead of list.</span>
+
+
+<span class="sd">II. READING DATA FROM GRID</span>
```
Comment (sgossage) [inline docs/v1.0.4/_modules/posydon/grids/psygrid.html:197]:
```suggestion
<span class="sd">II. READING DATA FROM A GRID</span>
```

**Comment by sgossage** [comment]
There are a number of typos throughout our documentation. I've noted a few above, but have not combed through everything. 

We could address these in a separate PR dedicated to fixing them though.

**Comment by maxbriel** [comment]
Thank you for going through this!
This is generated from the v1.0.4 codebase and the repeating typos probably originate from one source (or just a few of them for different repeats).
I will fix these since they have been found, but since it's v1 documentation I'm less concerned about catching them all since I think a lot of this documentation has changed in v2 :)
This PR is to make sure that the documentation version of v1 is up-to-date when we do the final v1 release as v1.0.4 before we merge development and v2 code to the main branch :)

**Comment by elizabethteng** [comment]
Hi Max, when looking through the new files you created for this PR, I noticed that a lot of them (those which have paths beginning with docs/v1.0.4/_modules/posydon/) have many lines that are identical across files. Is this not redundant? Since we didn't already have these in the docs, I'm not 100% sure how they are functioning.

**Comment by elizabethteng** [comment]
I have looked through all of the files other than most of the new files you added, addressed in the previous comment. (I also skipped the js and css files, as I am not familiar with those languages)

I had some small suggestions and questions but otherwise this PR looks great!

**Comment by mkruckow** [comment]
> Hi Max, when looking through the new files you created for this PR, I noticed that a lot of them (those which have paths beginning with docs/v1.0.4/_modules/posydon/) have many lines that are identical across files. Is this not redundant? Since we didn't already have these in the docs, I'm not 100% sure how they are functioning.

OK, maybe we should make sure, that any of us knows, what is going on here. We have two main parts: source files and production files. The source files are the ones we should change. The production files are generated from the source, thus they have a lot of common lines, which simply originate from the compilation. Thus, all changes should be done on the source files not on the production files (unless there are final changes of mistakes introduced by the compilation, still here it is better to change the source), because they will get replaced when you recompile the documentation.
Why do we have the production files in this PR? The reason is that those are the files which will be hosted on the webpage and in this PR we modify the v1 documentation, which is static/frozen (we don't have/want there an automatic compilation for any (accidental) change on the source files). All the production files are the main reason for github to think of many changes in this PR, while most of it is just replacing complied files.
P.S.: In contrast for the unstable documentation we only modify and keep the source files in the repo and have an automatic compilation on a merged PR.

**Comment by maxbriel** [comment]
To add to Matthias' comment with specific details:
- These files are generated using `sphinx` from the documentation files in [this PR](https://github.com/POSYDON-code/POSYDON/pull/450/files) on the POSYDON code github.
- It transforms `rst` files, `ipynb` files, and `docstrings` (The source files) into HTML pages for the website (the production files).
- Because it also creates a browsable structure (links of the left-side of the documentation website), some typos like Plot 1D and Plot 2D repeat in basically every file. This is because it's the title of one of the pages. Thus it repeats in every file with the sidebar links. Fixing this requires changes to the source file. ([See here for an example](https://github.com/POSYDON-code/POSYDON/pull/450/commits/8fea3f4765519c2c456bedaf6fbe5ee934df3228))

I should have explained this better to start with.
I have added the things you found here into [v1.0.4 PR](https://github.com/POSYDON-code/POSYDON/pull/450) and I've made [this PR](https://github.com/POSYDON-code/POSYDON/pull/453) to include it in the development branch.

**Review by sgossage — APPROVED** [review]

**Review by mkruckow — APPROVED** [review]
Before you merge this PR, could you please double check that all files are correct? I'm mainly concerned about the files, where github tells they are too big for a diff (especially an html file shouldn't be several MB in size without very good reason), e.g. [POSYDON-code.github.io/docs/v1.0.4/api/posydon.binary_evol.flow_chart.html](https://github.com/POSYDON-code/POSYDON-code.github.io/blob/max_v1.0.4_update/docs/v1.0.4/api/posydon.binary_evol.flow_chart.html)?

**Pull request merged**